lib/plugins/audit.js

```diff
@@ -196,68 +196,90 @@ function auditLogger(opts) {
     if (typeof printLog === 'undefined') {
         printLog = true;
     }
+
+    /**
+     * Serialize the HTTP Request into a plain object, for logging
+     *
+     * @private
+     * @function auditRequestSerializer
+     * @param {http.ClientRequest} req - Incoming request
+     * @returns {object} Serialized request
+     */
+    function auditRequestSerializer(req) {
+        if (!req) {
+            return false;
+        }
+
+        var timers = {};
+        (req.timers || []).forEach(function forEach(time) {
+            var t = time.time;
+            var _t = Math.floor(1000000 * t[0] + t[1] / 1000);
+            timers[time.name] = (timers[time.name] || 0) + _t;
+        });
+        return {
+            // account for native and queryParser plugin usage
+            query: typeof req.query === 'function' ? req.query() : req.query,
+            method: req.method,
+            url: req.url,
+            headers: req.headers,
+            httpVersion: req.httpVersion,
+            trailers: req.trailers,
+            version: req.version(),
+            body: opts.body === true ? req.body : undefined,
+            timers: timers,
+            connectionState: req.connectionState && req.connectionState()
+        };
+    }
+
+    /**
+     * Serialize the HTTP Response into a plain object, for logging
+     *
+     * @private
+     * @function auditResponseSerializer
+     * @param {http.ServerResponse} res - outgoing server response
+     * @returns {object} Serialized response
+     */
+    function auditResponseSerializer(res) {
+        if (!res) {
+            return false;
+        }
+
+        var body;
+
+        if (opts.body === true) {
+            if (res._body instanceof HttpError) {
+                body = res._body.body;
+            } else {
+                body = res._body;
+            }
+        }
+
+        return {
+            statusCode: res.statusCode,
+            headers: getResponseHeaders(res),
+            trailer: res._trailer || false,
+            body: body
+        };
+    }
+
     var errSerializer = bunyan.stdSerializers.err;
+    var reqSerializer = auditRequestSerializer;
+    var resSerializer = auditResponseSerializer;
 
-    if (opts.log.serializers && opts.log.serializers.err) {
-        errSerializer = opts.log.serializers.err;
+    // Allow override of serializers
+    if (opts.log.serializers) {
```

Comment:
I know this is how we were doing it before, but I wonder if we can be more explicit here, i.e., pass in opts.serializers as a top level option, rather than implicitly depending on any serializers on the existing logger. What do you think @hekike? If we went that route we might have to keep the old L201 line in there though for backwards compatibility.

Reply:
If my understanding is correct, you're thinking that we would use the entire serializers object instead of picking off three (req, res, err) and using those. This would allow consumers the flexibility of owning the serializers without having to keep adding conditional logic in this code.

I prefer this approach. However, to be fully backward compatible, we'll probably end up with the same conditional logic to ensure that the Restify default serializers still run if they're not explicitly passed by consumers.

Happy to make the change in either direction.

Reply:
Basically, I am thinking of an explicit serializers option:

```js
SERVER.once('after', restify.plugins.auditLogger({
    serializers: {...}
});
```

Then within the audit logger we'd pull serializers from a top level `opts.serializers` instead of from `opts.log.serializers`.  I believe the fallback logic you have for picking between a passed in serializer and the default one is still applicable. This just makes serializers a bit more explicit since we're creating a new child logger here anyway.